virtcontainers/qemu.go

```diff
@@ -628,7 +596,11 @@ func (q *qemu) togglePauseSandbox(pause bool) error {
 	return nil
 }
 
-func (q *qemu) qmpSetup() (*govmmQemu.QMP, error) {
+func (q *qemu) qmpSetup() error {
+	if q.qmpMonitorCh.qmp != nil {
+		return nil
```

Comment:
Just a thought here, do we need this to be idempotent? Is there any reason we would call qmpSetup() with `qmp` being not nil?

Reply:
`qmpSetup()` is now always called before any qmp talks. But `qmpShutdown()` is only called explicitly when qemu.disconnect() is called. So yes, there are lots of cases when qmpSetup() is called with qmp being not nil.